src/core/csv/reader_parsers.cc

```diff
@@ -284,7 +291,18 @@ void parse_float32_hex(FreadTokenizer& ctx) {
       else if (E == 126 && Eneg && acc) /* subnormal */ E = 0;
       else goto fail;
     } else {
+      #if DT_COMPILER_MSVC
+        #pragma warning(push)
+        // unary minus operator applied to unsigned type, result still unsigned
+        #pragma warning(disable : 4146) 
+      #endif
+      
       E = 127 + (Eneg? -E : E);
```

Comment:
I guess the question here is: do we want to modify the code (making it ugly sometimes for the purpose not clear to the reader) to get rid of warning _or_ we explicitly disable the warning writing the code in a normal usual way? On my last PR I thought you preferred the later option.

Reply:
I guess it's just the question of readability. In this case moving a constant around does not make the code any less readable, whereas adding ≈10 lines of compiler directives certainly creates lots of visual noise.

Reply:
I agree, but we just added "≈10 lines of compiler directives" for each `if ... xassert...` statement in the previous PR :) So I propose we converge here on the strategy for fixing this kind of warnings in future. 

Btw, for `if ... xassert ...` I found that it is enough just to do `if ... { xassert ... }` to silence the warning.

Reply:
You're right, that was not ideal either. Perhaps it would be just easier to modify the `xassert()` macro so that it expands into something trivial yet non-empty in non-debug mode? Like `(void)0` for example.

Reply:
Good idea. For the moment, if we use curly brackets even for one-liners we should have no warnings produced. From my point of view, curly brackets also improve readability.